File: python/templates/timer.py
# python/templates/timer.py
import time


class Timer:
    def __init__(self, limit_sec: float):
        self.start = time.perf_counter()
        self.limit = limit_sec

    def elapsed(self) -> float:
        return time.perf_counter() - self.start

    def remaining(self) -> float:
        return self.limit - self.elapsed()

    def expired(self) -> bool:
        return self.remaining() <= 0


if __name__ == "__main__":
    timer = Timer(1.0)
    while not timer.expired():
        pass
    print(f"elapsed: {timer.elapsed():.3f}s")

File: go/templates/timer.go
// go/templates/timer.go
package main

import (
	"fmt"
	"time"
)

type Timer struct {
	start time.Time
	limit time.Duration
}

func NewTimer(limitSec float64) *Timer {
	return &Timer{
		start: time.Now(),
		limit: time.Duration(limitSec * float64(time.Second)),
	}
}

func (t *Timer) Elapsed() time.Duration {
	return time.Since(t.start)
}

func (t *Timer) Remaining() time.Duration {
	return t.limit - t.Elapsed()
}

func (t *Timer) Expired() bool {
	return t.Remaining() <= 0
}

func main() {
	timer := NewTimer(0.01)
	for !timer.Expired() {
	}
	fmt.Printf("elapsed: %.3fs\n", timer.Elapsed().Seconds())
}
